app/backend/controllers/login-signup.js

```diff
@@ -86,6 +102,31 @@ module.exports.login = async (request, response) => {
   }
 }
 
+module.exports.google = async (request, response) => {
+  let User = request.models['User']
+  const googleId = request.body.googleId  //Email and password fields in request body
+
+  if (!googleId) {           // If there's no email field in the request, return status 400
+    response.status(400).send({ errmsg: 'Google ID required' })
+  } else{
+    try {
+      let userRegistered = await User.findOne({googleId})  // Retrieve the user instance from database
+      if (!userRegistered) {  // If no instance is returned, credentials are invalid
+        response.status(400).send({ errmsg: 'User must be signed up' })
+      }else{
+        request.session['user'] = userRegistered
+        const { _id, name, surname, email, location, isTrader, iban, tckn, isPublic, isVerified, googleId} = userRegistered  // Extract certain keys from user
+        response.send({
```

Comment:
Sending response twice causes error, I think it's a small flow issue here that can be fixed easily.